src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,10 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
+      if (!Properties.getControlAllCanalsBetweenTerritoriesToPass(data) && !failureMessage.isPresent()) {
```

Comment:
A comment stating how/why we are doing a 'match all' vs 'match any' would be of value. 

I'm finding the key problem is that we've buried the main logic flag in the guts of this function. 
Seems like it should be more like:
```
if( Properties.getControlAllCanalsBetweenTerritoriesToPass(data)) {
   canals.allMatch( canPass )
} else {
  canals.anyMatch( canPass)
}
```

Pls fix this up one way or another so it can be understood more readily.